mcts/src/main/java/org/motechproject/nms/mcts/service/impl/MctsWsImportServiceImpl.java

```diff
@@ -106,24 +116,36 @@ public void importFromMcts(List<Long> stateIds, LocalDate referenceDate, URL end
         }
 
         for (Long stateId : stateIds) {
-            LOGGER.debug("Sending import message for stateId {}", stateId);
-            Map<String, Object> eventParams = new HashMap<>();
-            eventParams.put(Constants.DATE_PARAM, referenceDate);
-            eventParams.put(Constants.STATE_ID_PARAM, stateId);
-            eventParams.put(Constants.ENDPOINT_PARAM, endpoint);
-            eventRelay.sendEventMessage(new MotechEvent(Constants.MCTS_MOTHER_IMPORT_SUBJECT, eventParams));
-            eventRelay.sendEventMessage(new MotechEvent(Constants.MCTS_CHILD_IMPORT_SUBJECT, eventParams));
-            eventRelay.sendEventMessage(new MotechEvent(Constants.MCTS_ASHA_IMPORT_SUBJECT, eventParams));
+            sendImportEventForAUserType(stateId, MctsUserType.MOTHER, referenceDate, endpoint, Constants.MCTS_MOTHER_IMPORT_SUBJECT);
+            sendImportEventForAUserType(stateId, MctsUserType.CHILD, referenceDate, endpoint, Constants.MCTS_CHILD_IMPORT_SUBJECT);
+            sendImportEventForAUserType(stateId, MctsUserType.ASHA, referenceDate, endpoint, Constants.MCTS_ASHA_IMPORT_SUBJECT);
         }
 
         LOGGER.info("Initiated import workflow from MCTS for mothers, children and ashas");
     }
 
+    private void sendImportEventForAUserType(Long stateId, MctsUserType userType, LocalDate referenceDate, URL endpoint, String importSubject) {
+
+        LOGGER.debug("Fetching all the failed imports in the last 7days for stateId {} and UserType {}", stateId, userType);
+        QueryParams queryParams = new QueryParams(new Order("importDate", Order.Direction.ASC));
+        List<MctsImportFailRecord> failedImports = mctsImportFailRecordDataService.getByStateAndImportdateAndUsertype(stateId, referenceDate.minusDays(6), userType, queryParams);
+        LocalDate startDate = failedImports.isEmpty() ? referenceDate : failedImports.get(0).getImportDate();
```

Comment:
Are you sure that the previous query always returns back an ordered list? get(0) looks dangerous